### DISCUSSION TITLE
Do not require the number of `"`s at the start and end token in a raw string literal being the same if the literal is single-line

### DISCUSSION THREAD

**Discussion by kyoyama-kazusa** [dotnet/roslyn#60919] [General] [answered]
Today, the feature "Raw String Literals" is very nice to operate and display a string value. However, a raw string literal must require the same count of quotes (`"`) at the start and the end string token:

```csharp
string s = """hello!"""; // OK
string t = """hello, "name""""; // Compile error.
```

We must use multi-line raw string literal instead to avoid the compile error:

```csharp
string t = """
    hello, "name"
    """;
```

or use verbatim strings:

```csharp
string t = @"hello, ""name""";
```

But in this case each quote `"` will be repeated once inside the string literal.

I think that it will be very nice when the compiler do not require the same count of `"`s at the start and the end token, and the side that uses exceeded quotes `"` will be treated as the normal quote mark inside the literal:

```csharp
string s = """Inside quote: "something"""";
// equivalent to
string s = "Inside quote: \"something\"";
string s = @"Inside quote: ""something""";
string s = """
    Inside quote: "something"
    """;
```

Another example:

```csharp
string t = """"blah blah", said him.""";
// equivalent to
string t = "\"blah blah\", said him.";
string t = @"""blah blah"", said him.";
string t = """
    "blah blah", said him.
    """;
```

rather than a compiler error message.

To be more excessively, quote-only raw string literals will still be disallowed:

```csharp
string s = """"""; // Still not allowed
string s = """"""""; // Still not allowed
```

Any considerations on it? Thank you.

**Comment by CyrusNajmabadi** [comment, marked as answer]
This will only help with strings that end with quotes. 

That's too narrow a case to make it worth making the feature more complex here. Strings that start or end with a quote at easily handled by just using the mutiline form.

**Reply by kyoyama-kazusa** [reply]
Thanks for your reply. That sounds reasonable. 😄
